src/main/java/nl/tudelft/lifetiles/graph/view/TileView.java

```diff
@@ -171,9 +171,9 @@ private boolean segmentFree(final int ind, final SequenceSegment segment) {
     private void segmentInsert(final int index, final SequenceSegment segment) {
         for (int height = 0; height < segment.getSources().size(); height++) {
             if (index + height < lanes.size()) {
-                lanes.set(index + height, segment.getEnd());
+                lanes.set(index + height, segment.getUnifiedEnd());
```

Comment:
Double code, you should make `index + height, segment.getUnifiedEnd()` a variable outside the condition scope.